app/src/main/java/com/tsu/calculator/MainModel.kt

```diff
@@ -1,35 +1,185 @@
+    private fun calculate(first: Double, second: Double, operator: Buttons): Double {
         val result: Double
 
         when (operator) {
-            "divide" -> {
+            Buttons.DIVIDE -> {
                 if (second.equals(0.0)) {
-                    throw Exception("Division by zero")
+                    throw Exception(DIVISION_ERROR)
                 } else {
                     result = first.div(second)
                 }
             }
-            "multiply" -> {
+            Buttons.MULTIPLY -> {
                 result = first * second
             }
-            "subtract" -> {
+            Buttons.MINUS -> {
                 result = first - second
             }
-            "add" -> {
+            Buttons.PLUS -> {
                 result = first + second
             }
             else -> {
-                throw Exception("Unknown operator")
+                Log.e("Model", OPERATOR_ERROR + " " + operator.label)
+                throw Exception(OPERATOR_ERROR)
             }
         }
 
         return result
     }
+
+    fun getResult(): String {
+        val operation = _operation
+        if (operation == null) {
+            if (_firstOperand.isEmpty()) {
+                _firstOperand = _secondOperand
+                clearSecondOperand()
+                return ""
+            } else {
+                clear()
+                return ERROR
+            }
+        }
+
+        if (_firstOperand.isNotEmpty() && _secondOperand.isNotEmpty()) {
+            val firstNumber = _firstOperand.toDouble()
+            val secondNumber = _secondOperand.toDouble()
+
+            try {
+                val result = calculate(firstNumber, secondNumber, operation)
+                _firstOperand = result.toString()
+                clearSecondOperand()
+                return result.toString()
+            } catch (e: Exception) {
+                clear()
+                return ERROR
+            }
+        }
+
+        _firstOperand = _secondOperand
+        clearSecondOperand()
+        return ""
+    }
+
+    fun setOperation(operation: Buttons): String {
+        if (_firstOperand.isEmpty() && _secondOperand.isEmpty()) {
+            _firstOperand = "0"
+        }
+
+        if (_secondOperand.isNotEmpty()) {
+            val result = getResult()
+            _operation = operation
+            return result
+        }
+
+        _operation = operation
+        return ""
+    }
+
+    private fun clearSecondOperand() {
+        commaAdded = false
+        _secondOperand = ""
+    }
+
+    fun clear() {
+        _firstOperand = ""
+        clearSecondOperand()
+        _operation = null
+    }
+
+    fun changeSign(): String {
+        if ((_secondOperand.isEmpty() || _secondOperand == "0") &&
+            (_firstOperand.isEmpty() || _firstOperand == "0")
+        ) {
+            return ""
+        }
+
+        if (_secondOperand.isEmpty() || _secondOperand == "0") {
+            if (_firstOperand[0] == '-') {
```

Comment:
Poor practice to compare strings. You should use enum classes instead

For example your enum class could contain an operation string (such you did in in `Buttons` enum). Then if you need to get string value from `enum` you could call `Buttons.label`. If you need to get `enum class` from string you could call `Buttons.valueOf(string)`